src/context/Auth.tsx

```diff
@@ -1,27 +1,44 @@
 import FullPageSpinner from 'components/fullPage/Spinner';
+import { fromUnixTime } from 'date-fns';
 import isFuture from 'date-fns/isFuture';
+import { AuthTokenResponseReduced } from 'endpoints/auth';
 import { useAsync } from 'hooks/useAsync';
+import { isEmpty } from 'lodash';
 import React, { useCallback, useMemo } from 'react';
 import FullPageError from '../components/fullPage/Error';
 import { auth } from '../services/auth';
 
 export interface AuthContextType {
     login: (username: string) => Promise<void>;
     logout: () => Promise<void>;
-    user: string | null;
+    user: AuthTokenResponseReduced | null;
+}
+
+export function checkTokens(tokens: AuthTokenResponseReduced) {
+    let response;
+
+    if (isFuture(fromUnixTime(tokens.expires))) {
+        response = tokens;
+    } else {
+        response = null;
+    }
```

Comment:
Need to check if the token expired. The new endpoint returns unit time so using the handy date-fns library again.